### PR TITLE
feat(vitals): Update vitals cards and landing with three states

### PR DESCRIPTION
**Pull request by k-fish**
### Summary
This modifies cards and passing/failing references to vitals to be good/meh/poor instead, with updated colors to match. Added colorBar as a feature component for now, with a storybook for it.

The x/y passed views is replaced with a query which aggregates all good/meh/poor counts across all the vitals to give an overview on transaction summary.

#### Todo
- Fix multiseries hack
- Double check feature flags
- Add/modify tests

### CODE REVIEW

**Comment by k-fish** [comment]
Putting this up for review, still have a bit left to do on it. Vital card will probably need a bit of work in the future, but I'll probably wait until backend cards to update it since I'm not 100% what they'll look like.

File: src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx [hunk 1]
```diff
@@ -1,28 +1,81 @@
...
+  [WebVital.FID]: `count_at_least(measurements.fid, 300)`,
+  [WebVital.CLS]: `count_at_least(measurements.cls, 0.25)`,
+};
+
+export const vitalsMehFields = {
+  [WebVital.FP]: `count_at_least(measurements.fp, 1000)`,
+  [WebVital.FCP]: `count_at_least(measurements.fcp, 1000)`,
+  [WebVital.LCP]: `count_at_least(measurements.lcp, 2500)`,
+  [WebVital.FID]: `count_at_least(measurements.fid, 100)`,
+  [WebVital.CLS]: `count_at_least(measurements.cls, 0.1)`,
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:42]: Should we think about adding a new discover function that will help use have smaller requests? Perhaps a function that returns a list with the `[bad, meh, good]` results?

File: src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx [hunk 2]
```diff
@@ -1,28 +1,81 @@
...
+  [WebVital.FCP]: 'p75(measurements.fcp)',
+  [WebVital.LCP]: 'p75(measurements.lcp)',
+  [WebVital.FID]: 'p75(measurements.fid)',
+  [WebVital.CLS]: 'p75(measurements.cls)',
+};
+
+export enum VitalState {
+  POOR = 'Poor',
+  MEH = 'Meh',
+  GOOD = 'Good',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:64]: Do these need translations?

File: src/sentry/static/sentry/app/views/performance/vitalDetail/vitalChart.tsx [hunk 3]
```diff
@@ -136,11 +138,34 @@ class VitalChart extends React.Component<Props> {
...
+          silent: true,
+          lineStyle: {
+            color: theme.textColor,
+            type: 'dashed',
+            width: 1,
+          },
+          label: {
+            show: true,
+            position: 'insideEndBottom',
+            formatter: 'Meh',
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/vitalDetail/vitalChart.tsx:164]:
```suggestion
            formatter: t('Meh'),
```

File: src/sentry/static/sentry/app/views/performance/vitalDetail/vitalChart.tsx [hunk 4]
```diff
@@ -204,14 +229,20 @@ class VitalChart extends React.Component<Props> {
...
                     <ErrorPanel>
                       <IconWarning color="gray500" size="lg" />
                     </ErrorPanel>
                   );
                 }
+
+                const singleSeries = _results?.find(
+                  r => r.seriesName === `p75(${vitalName})`
+                );
+                const results = _results ? (singleSeries ? [singleSeries] : []) : []; // TODO: Fix this multiseries hack
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/performance/vitalDetail/vitalChart.tsx:244]: There is a function in utils/charts that lets you separate single/multi-series result shapes with typing.

File: src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx (hunk 1 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:42]: I've thought about building a special query for the cards, Will was looking at it.

Comment (wmak) [reply src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:42]: :+1: Totally agreed, Shift some of these calls with nearly 20 fields to special endpoints will help avoid other issues we're having too

Comment (markstory) [reply src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:42]: We can and should ship this. A specific endpoint would be ideal, and would let us optimize loading data and ensure vitals always look the same in each place they show up.

File: src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx (hunk 2 above)
Comment (k-fish) [reply src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx:64]: translating `meh` might be tricky 😄

**Review by markstory — APPROVED** [review]

**Pull request merged**